### PR TITLE
Fix: Feed notification

### PR DESCRIPTION
**Pull request by robert-schardt**
## What

Change notification-channel for notify-mattermost-2nd-gen.yml
Adapt if conditions when using notify-mattermost* workflows

## Why

Currently https://github.com/greenbone/workflows/blob/main/.github/workflows/notify-mattermost-2nd-gen.yml#L36 notifies in channel: [PD:2ndGenDeployment](https://mattermost.greenbone.net/gb/channels/pd2ndgendeployment) but should be [TOD:FeedDeployment:Notification](https://mattermost.greenbone.net/gb/channels/pdfeeddeploymentnotification) 

Also check and adapt the if conditions for all workflows in greenbone/workflows using the notify-mattermost-2nd.gen workflow -> They should be run on every build.

## References

https://jira.greenbone.net/browse/DEVOPS-1254

### CODE REVIEW

File: .github/workflows/notify-mattermost-2nd-gen.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
         uses: greenbone/actions/mattermost-notify@v3
         with:
           url: ${{ secrets.MATTERMOST_WEBHOOK_URL }}
-          channel: "pd2ndgendeployment"
+          channel: "todfeeddeploymentnotification"
```
[the remark this answers is not in the document]
Comment (robert-schardt) [reply .github/workflows/notify-mattermost-2nd-gen.yml:36]: should we introduce an optional parameter?

[the remark this answers is not in the document]
Comment (y0urself) [reply .github/workflows/notify-mattermost-2nd-gen.yml:36]: yes please

[the remark this answers is not in the document]
Comment (robert-schardt) [reply .github/workflows/notify-mattermost-2nd-gen.yml]: @y0urself What would be wiser to set as default then?
Are there more components using todfeeddeployment or more that use pd2ndgendeployment?

File: .github/workflows/container-build-push-2nd-gen.yml [hunk 2]
```diff
@@ -38,6 +38,10 @@ on:
         description: "The artifact path to extract the artifact into."
         required: false
         type: string
+      notification-channel:
+        description: "The name of the channel where the notification happens."
+        type: string
+        default: "todfeeddeploymentnotification"
```
Comment (y0urself) [inline .github/workflows/container-build-push-2nd-gen.yml:44]:
```suggestion
        default: "pd2ndgendeployment"
```

File: .github/workflows/notify-mattermost-2nd-gen.yml [hunk 3]
```diff
@@ -19,6 +19,10 @@ on:
         description: "The monitored job, job status."
         type: string
         required: true
+      channel:
+        description: "The name of the channel where the notification happens."
+        type: string
+        required: true
```
Comment (y0urself) [inline .github/workflows/notify-mattermost-2nd-gen.yml:25]:
```suggestion
        description: "The name of the channel where the notification happens."
        type: string
        default: pd2ndgendeployment
```

**Review by y0urself — COMMENTED** [review]
we can do a default here too, as the workflow is named 2nd gen ... it only should notify 2nd gen ...?

**Comment by robert-schardt** [comment]
Searched through all the feed repositories, only greenbone-feed-sync/.github/workflows/push.yml might need adaption.

File: .github/workflows/container-build-push-feed.yml [hunk 4]
```diff
@@ -42,6 +42,10 @@ on:
         description: "Is this the vts image?"
         default: "false"
         type: string
+      notification-channel:
+        description: "The name of the channel where the notification happens."
+        type: string
+        default: "toddevopsgithubstatusnotification"
```
Comment (y0urself) [inline .github/workflows/container-build-push-feed.yml:48]:
```suggestion
        default: "pdfeeddeploymentnotification"
```

**Review by y0urself — APPROVED** [review]

**Pull request merged**